host/frontend/host-orchestrator/intercept/js/server_connector.js

```diff
@@ -29,8 +29,16 @@
 // the client loaded.
 // In this case the device id is passed as a parameter in the url.
 export function deviceId() {
-  const urlParams = new URLSearchParams(window.location.search);
-  return urlParams.get('deviceId');
+  // The server connector is loaded from the client.html file, which is loaded
+  // with a path like "/devices/{deviceId}/files/"
+  let path = location.pathname;
+  let beginIdx = path.indexOf("/devices/") + "/devices/".length;
+  let endIdx = path.indexOf("/files/");
+  if (beginIdx < 0 || endIdx < 0) {
+    // The path doesn't match our expectations
+    throw "server connector is incompatible with this server";
```

Comment:
Can this throw an [Error instance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) rather than a string?

Reply:
Done